### PR TITLE
Feat: Add support for response_format field in OpenAI Spec

### LINKED ISSUE #196: Add support for `response_format` in OpenAI Spec
## 🚀 Feature
Add support for `response_format` in OpenAI Spec

### Motivation
- [Structured Outputs](https://openai.com/index/introducing-structured-outputs-in-the-api/)

### PR DESCRIPTION
**Pull request by bhimrazy**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes #196 .

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/litserve/specs/openai.py [hunk 1]
```diff
@@ -105,6 +105,31 @@ class ToolCall(BaseModel):
...
+
+
+class ResponseFormatJSONObject(BaseModel):
+    type: Literal["json_object"]
+
+
+class JSONSchema(BaseModel):
+    name: str
+    description: Optional[str] = None
+    schema: Optional[Dict[str, object]] = None
```
Comment (bhimrazy) [inline src/litserve/specs/openai.py:119]: Hi @aniketmaurya ,

I'm encountering the following warning, due to a field in JSONSchema:

*Warning: Field name 'schema' in 'JSONSchema' shadows an attribute in parent 'BaseModel'.*

I would appreciate any suggestions you might have for resolving this issue. Is there a preferred way to handle such cases?

Thank you for your help!

Comment (aniketmaurya) [reply src/litserve/specs/openai.py:119]: hi @bhimrazy, sorry for late response. somehow I missed the ping. Have you tried it with [alias](https://stackoverflow.com/questions/76728740/two-fields-with-same-name-causing-problem-in-pydantic-model-classes) in Pydantic?

**Review by aniketmaurya — COMMENTED** [review]
Looks great so far @bhimrazy!! Let's check the Pydantic alias to avoid shadowing the BaseModel attribute.

File: src/litserve/specs/openai.py (hunk 1 above)
Comment (bhimrazy) [reply src/litserve/specs/openai.py:119]: No worries @aniketmaurya and Thanks for the suggestion I hadn't considered using an alias. I'll give that a try and see if it resolves the warning. Appreciate the help! 🙂

**Review by aniketmaurya — APPROVED** [review]

**Pull request merged**